plugin/cog/cog-reader/src/test/java/it/geosolutions/imageioimpl/plugins/cog/azure/CogImageReaderAzureIT.java

```diff
@@ -0,0 +1,216 @@
+ * {@link AzuriteContainer}.
+ *
+ * <p>This test uses the Azurite emulator to test Azure Blob Storage integration without requiring real Azure
+ * credentials or making actual network calls to Azure.
+ *
+ * <p>The emulator is configured via system properties that are read by {@link AzureConfigurationProperties} via
+ * {@code PropertyLocator}:
+ *
+ * <ul>
+ *   <li>{@code azure.reader.serviceurl} - The Azurite emulator endpoint (e.g., http://localhost:port/accountName)
+ *   <li>{@code azure.reader.accountname} - The Azurite default account name (devstoreaccount1)
+ *   <li>{@code azure.reader.accountkey} - The Azurite default account key
+ * </ul>
+ *
+ * <p>Test files are uploaded to the emulator using the Azure Storage Blob SDK, and then accessed via
+ * {@link AzureRangeReader} using standard HTTP URLs pointing to the emulator.
+ *
+ * <p><b>Note:</b> Property names must be lowercase (e.g., {@code azure.reader.serviceurl} not
+ * {@code azure.reader.serviceURL}) to work correctly with {@code PropertyLocator.getEnvironmentValue()}.
+ */
+public class CogImageReaderAzureIT extends BaseCogImageReaderTest {
+
+    private static final DockerImageName DOCKER_IMAGE_NAME =
+            DockerImageName.parse("mcr.microsoft.com/azure-storage/azurite:3.35.0");
+
+    private static final String CONTAINER_NAME = "test-container";
+
+    @ClassRule
+    public static CogTestData testData = new CogTestData();
+
+    public static AzuriteContainer container = new AzuriteContainer(DOCKER_IMAGE_NAME);
+    // These are the default account name and key used by Azurite
+    private static final String ACCOUNT_NAME = "devstoreaccount1";
+    private static final String ACCOUNT_KEY =
+            "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+
+    /**
+     * Skip tests when caching is enabled due to ArrayIndexOutOfBoundsException in CachingCogImageInputStream.read():
+     *
+     * <pre>
+     * java.lang.ArrayIndexOutOfBoundsException: arraycopy: last source index 22366 out of bounds for byte[16384]
+     * </pre>
+     */
+    @Before
+    public void skipCachingTests() {
+        Assume.assumeFalse(
+                "Caching tests are skipped due to known issue with CachingCogImageInputStream", super.caching);
+    }
+
+    @BeforeClass
+    public static void setUpContainer() throws IOException {
+        container.start();
+
+        // Configure Azure client to use Azurite
+        // Azurite service URL includes the account name in the path
+        // Note: Property names must be lowercase to work with
+        // PropertyLocator.getEnvironmentValue()
+        Integer port = container.getMappedPort(10000);
+        String azuriteEndpoint = "http://localhost:" + port + "/" + ACCOUNT_NAME;
+        System.setProperty("azure.reader.serviceurl", azuriteEndpoint);
+        System.setProperty("azure.reader.accountname", ACCOUNT_NAME);
+        System.setProperty("azure.reader.accountkey", ACCOUNT_KEY);
```

Comment:
Since test JVMs are reused, setting system variables without clearing them might cause cross-test contamination in the future, I recommend clearing them in @AfterClass.